main.go

```diff
@@ -202,9 +212,12 @@ func main() {
 	managerOpts.AddToManager = addToManager
 
 	ctx := pkgcfg.WithConfig(defaultConfig)
-	if defaultConfig.Features.UnifiedStorageQuota {
+	if defaultConfig.Features.UnifiedStorageQuota || defaultConfig.Features.WorkloadDomainIsolation {
 		ctx = cource.WithContext(ctx)
 	}
+	if defaultConfig.Features.WorkloadDomainIsolation {
+		ctx = watcher.WithContext(ctx)
+	}
 	initFeaturesFromCapabilities(ctx, setupLog)
```

Comment:
I think we'd want to update the features w/ the capabilities before decided on what is enabled.

That said, it might just be easier to always init the `cource` and `watcher` in the context here so it is always there, and for a world where these things are flipped at runtime. It should be pretty low risk.